wgpu-core/src/device/mod.rs

```diff
@@ -1398,15 +1400,24 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         device_id: id::DeviceId,
         desc: &binding_model::BindGroupDescriptor,
         id_in: Input<G, id::BindGroupId>,
-    ) -> id::BindGroupId {
+    ) -> Result<id::BindGroupId, BindGroupError> {
+        use crate::binding_model::BindingResource;
```

Comment:
perhaps use a shorthand like `as Br`? The convention so far for `device` module is to use fully qualified paths to things, like `binding_model::BindingResource`, but since you are matching a lot locally, you can have a shortcut. We shouldn't stop half-way :)